stan/math/fwd/core/std_complex.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+#define STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+
+#include <stan/math/prim/core/complex_base.hpp>
+#include <stan/math/fwd/core/fvar.hpp>
+#include <complex>
+#include <type_traits>
+
+namespace std {
+
+/**
+ * Specialization of the standard libary complex number type for
+ * reverse-mode autodiff type `stan::math::fvar<T>`.
+ */
+template <typename T>
+class complex<stan::math::fvar<T>>
+    : public stan::math::complex_base<stan::math::fvar<T>> {
```

Comment:
If this is a crtp should this be

```
public stan::math::complex_base<complex<stan::math::fvar<T>>>
```

So the type is recurrent in the base class?

Reply:
No.  The `complex` (which would have to be `std::complex`) is redundant.  I only take the value type `T` as the template parameter, but the "derived" class is `std::complex<T>`.  If I suppoied all of `complex<fvar<T>>`, then I'd just have to go fishing the `T` out with traits again.

Reply:
It is definitely redundant, though if I use a pattern I like keeping it standard instead of 'ish. But whether a strict CRTP pattern here would be better is totally subjective. 

> then I'd just have to go fishing the T out with traits again.

Would argue that type traits change here would be fishing in a barrel

```cpp
  using value_type = value_type_t<V>;

  /**
   * Derived complex type used for function return types.
   */
  using complex_type = V;
```